viewer/static/js/app/components/containers/app-container.js

```diff
@@ -26,10 +26,12 @@ class AppContainer extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
+    const previousDatasetId = this.datasetId;
```

Comment:
You might be right. I'm having a hard time wrapping my mind around this at the moment given the inter-dependencies in *app-container* but feel free to follow up with the change you have in mind.